data/helpers.d/mysql

```diff
@@ -90,3 +90,47 @@ ynh_mysql_create_user() {
 ynh_mysql_drop_user() {
     ynh_mysql_execute_as_root "DROP USER '${1}'@'localhost';"
 }
+
+# Create a database, an user and its password. Then store the password in the app's config
+#
+# User of database will be store in db_user's variable.
+# Name of database will be store in db_name's variable.
+# And password in db_pwd's variable.
+#
+# usage: ynh_mysql_generate_db user name
+# | arg: user - Owner of the database
+# | arg: name - Name of the database
+ynh_mysql_generate_db () {
+	db_pwd=$(ynh_string_random)	# Generate a random password
+	ynh_mysql_create_db "$2" "$1" "$db_pwd"	# Create the database
+	ynh_app_setting_set $app mysqlpwd $db_pwd	# Store the password in the app's config
```

Comment:
I don't know much about the use case for this, but to me using `ynh_app_setting_set` looks like the right way to do. In fact we should document this setting (a few lines higher) as the way to get the mysql password if needed, instead of setting `db_pwd` (which won't exist in the parent if it's called in a subshell ?).